### PR TITLE
[Dy2St] Update limitations docs and refine styles

### PR DESCRIPTION
**Pull request by SigureMo**
主要是更新 PIR 相关内容，由于 PIR 相比于老 IR 会更加严格，因此更新了部分报错改写指南，另外部分原理相关内容也替换为 PIR 下的概念

- preview: http://preview-pr-7247.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/guides/jit/index_cn.html

### CODE REVIEW

File: docs/guides/jit/basic_usage_cn.md [hunk 1]
```diff
@@ -216,16 +216,16 @@ loader = paddle.io.DataLoader(dataset,
...
-+ 方式一在组网代码中 ``forward()`` 函数定义处装饰，意指将其转为静态图执行模式；
++ 方式一在组网代码中 `forward()` 函数定义处装饰，意指将其转为静态图执行模式；
 
-+ 方式二在组网代码后，对整个网络 ``Layer`` 对象调用了 ``paddle.jit.to_static()`` 方法，默认也是对 ``Layer.forward()`` 进行动静转换；
++ 方式二在组网代码后，对整个网络 `Layer` 对象调用了 `paddle.jit.to_static()` 方法，默认也是对 `Layer.forward()` 进行动静转换；
 
 + 两种方式均实现了动转静训练，最终实现的效果是一样。
 
 
-**值得注意的是：请确保被装饰的 ``Layer.forward`` 方法中仅实现预测功能，避免将训练所需的 loss 计算逻辑写入 forward 方法。**
```
Comment (DrRyanHuang) [inline docs/guides/jit/basic_usage_cn.md:226]: 要不要把 `仅实现预测功能` 修改为 `仅实现前向传播`

File: docs/guides/jit/basic_usage_cn.md [hunk 2]
```diff
@@ -304,29 +304,28 @@ class LinearNet(nn.Layer):
...
 1. 先执行了动转静。当然如果前面已经执行了动转静训练，则跳过这一步。在处理逻辑上，主要包含两个主要模块：
-    + 模型结构层面：将动态图模型中被 ``@paddle.jit.to_static`` 装饰的函数转化为完整的静态图 Program。
 
-    + 模型参数层面：将动态图模型中的参数（Parameters 和 Buffers ）转为 ``Persistable=True``  的静态图模型参数 Variable。
+    + 模型结构层面：将动态图模型中被 `@paddle.jit.to_static` 装饰的函数转化为完整的静态图 Program。
+    + 模型参数层面：将动态图模型中的参数（`Parameters` 和 `Buffers`）转为 `persistable=True` 的静态图模型参数 Variable。
 
-2. 再将静态图模型和参数导出为磁盘文件。Program 和 Variable 都可以直接序列化导出为磁盘文件，与前端代码完全解耦，导出的文件包括：
-    + 后缀为 ``.json`` 的模型结构文件；
+3. 再将静态图模型和参数导出为磁盘文件。`Program` 和 `Value` 都可以直接序列化导出为磁盘文件，与前端代码完全解耦，导出的文件包括：
```
Comment (DrRyanHuang) [inline docs/guides/jit/basic_usage_cn.md]: 上面👆🏻，静态图模型参数是 Variable 还是 Value 呢?

```suggestion
2. 再将静态图模型和参数导出为磁盘文件。`Program` 和 `Value` 都可以直接序列化导出为磁盘文件，与前端代码完全解耦，导出的文件包括：
```

File: docs/guides/jit/basic_usage_cn.md [hunk 3]
```diff
@@ -355,16 +354,16 @@ linear.pdiparams.info   // 存放和参数状态有关的额外信息
...
-+ **使用 ``paddle.jit.load`` 加载**：该方式可以载入模型结构和参数，传入数据即可训练或推理。
-+ **使用 ``paddle.load`` 加载**：如果已有组网代码，则只传入模型参数也可再训练，因此也可以选择该方式加载。
++ **使用 `paddle.jit.load` 加载**：该方式可以载入模型结构和参数，传入数据即可训练或推理。
++ **使用 `paddle.load` 加载**：如果已有组网代码，则只传入模型参数也可再训练，因此也可以选择该方式加载。
 
-##### 3.2.2.1 使用 ``paddle.jit.load`` 加载
+##### 3.2.2.1 使用 `paddle.jit.load` 加载
 
-使用 ``paddle.jit.load`` 载入，载入后得到的是一个 Layer 的派生类对象 TranslatedLayer ， TranslatedLayer 具有 Layer 的通用特征，可以进行模型调优。
+使用 `paddle.jit.load` 载入，载入后得到的是一个 Layer 的派生类对象 TranslatedLayer ， TranslatedLayer 具有 Layer 的通用特征，可以进行模型调优。
```
Comment (DrRyanHuang) [inline docs/guides/jit/basic_usage_cn.md]: `可以进行模型调优` 这个意思就是模型训练、微调吧? 还是说可以进行一些超参数的调整？要改成训练调优吗

File: docs/guides/jit/basic_usage_cn.md [hunk 4]
```diff
@@ -694,27 +693,27 @@ pred = loaded_layer(x)
     paddle.jit.save(layer, path)
     ```
 
-    > 注：只有在 forward 之外还需要保存其他函数时才用这个特性，如果仅装饰非 forward 函数，而 forward 本身函数没有被装饰，是不符合规范的。当保存多个函数时， ``InputSpec`` 信息需要在各个函数的 ``@paddle.jit.to_static`` 里分别指定，并且 ``input_spec`` 参数必须为 None，因为此时 save 接口 input_spec 参数无法知道它应该配置给哪个函数。
+    > 注：只有在 forward 之外还需要保存其他函数时才用这个特性，如果仅装饰非 forward 函数，而 forward 本身函数没有被装饰，是不符合规范的。当保存多个函数时， `InputSpec` 信息需要在各个函数的 `@paddle.jit.to_static` 里分别指定，并且 `input_spec` 参数必须为 None，因为此时 save 接口 input_spec 参数无法知道它应该配置给哪个函数。
```
Comment (DrRyanHuang) [inline docs/guides/jit/basic_usage_cn.md]: 这部分是不是和代码有些出入
input_spec参数是None呢? 还是应该是一个列表呢?

File: docs/guides/jit/basic_usage_cn.md [hunk 5]
```diff
@@ -883,12 +882,12 @@ class SimpleNet(Layer):
...
+其中 input_spec 参数是长度为 2 的 list ，对应 forward 函数的 x 和 bias_info 两个参数。 input_spec 的最后一个元素是包含键名为 x 的 `InputSpec` 对象的 dict ，对应参数 bias_info 的 Tensor 签名信息。
 
 
 **（4）方式四：指定非 Tensor 参数类型**
 
-若被装饰函数的参数列表除了 Tensor 类型，还包含其他如 int、 str 等非 Tensor 类型时，推荐在函数中使用 kwargs 形式定义非 Tensor 参数，如下述样例中的 ``use_act`` 参数。
+若被装饰函数的参数列表除了 Tensor 类型，还包含其他如 int、 str 等非 Tensor 类型时，推荐在函数中使用 kwargs 形式定义非 Tensor 参数，如下述样例中的 `use_act` 参数。
 
 ```python
 class SimpleNet(Layer):
```
Comment (DrRyanHuang) [inline docs/guides/jit/basic_usage_cn.md:890]: 下面的
```python
# 方式一：save inference model with use_act=False
net = to_static(input_spec=[InputSpec(shape=[None, 10], name='x')])
paddle.jit.save(net, path='./simple_net')
# 方式二：save inference model with use_act=True
net = to_static(input_spec=[InputSpec(shape=[None, 10], name='x'), True])
paddle.jit.save(net, path='./simple_net')
```
这部分是不是应该修改一下，把参数传给 `save` 或者，给 `to_static` 加参数

File: docs/guides/jit/case_analysis_cn.md [hunk 6]
```diff
@@ -32,24 +32,24 @@
...
-+ 建议模型搭建时，尽量考虑将预测主逻辑放到 ``forward`` 函数中
-    + 将训练独有的逻辑放到 子函数 中，通过 ``if self.training`` 来控制
++ 建议模型搭建时，尽量考虑将预测主逻辑放到 `forward` 函数中
+    + 将训练独有的逻辑放到 子函数 中，通过 `if self.training` 来控制
     + 最大程度抽离 **训练和预测** 的逻辑为 **公共子函数**
 
 
 ## 二、何时指定 InputSpec?
 
-在动转静的原理介绍中，静态图 ``Program`` 的生成需要依赖 ``Placeholder`` 信息，此信息可通过两种方式获得：
```
Comment (DrRyanHuang) [inline docs/guides/jit/case_analysis_cn.md:49]: 要不要把 `Placeholder` 直接替换为占位符呢?

File: docs/guides/jit/case_analysis_cn.md [hunk 7]
```diff
@@ -183,19 +183,19 @@ class SimpleNet(paddle.nn.Layer):
...
++ 推荐尽量放到 `__init__` 函数中一次性进行初始化
 
 
 ## 五、 建议都继承 nn.Layer
 
 
-动态图模型常常包含很多嵌套的子网络，建议各个自定义的子网络 ``sublayer`` **无论是否包含了参数，都继承 ``nn.Layer`` **。
+动态图模型常常包含很多嵌套的子网络，建议各个自定义的子网络 `sublayer` **无论是否包含了参数，都继承 `nn.Layer` **。
 
-从 **Parameters 和 Buffers**  章节可知，有些 ``paddle.to_tensor`` 接口转来的 ``Tensor`` 也可能参与预测逻辑分支的计算，即模型导出时，也需要作为参数序列化保存到 ``.pdiparams`` 文件中。
```
Comment (DrRyanHuang) [inline docs/guides/jit/case_analysis_cn.md:196]: > 从 **Parameters 和 Buffers** 章节

这个章节的超链接在哪里🤪

File: docs/guides/jit/case_analysis_cn.md [hunk 8]
```diff
@@ -269,9 +269,9 @@ jit.save(mode, model_path)
 ```
 
 
-推荐使用 ``self.training`` 或其他非 Tensor 类型的 bool 值进行区分。
+推荐使用 `self.training` 或其他非 Tensor 类型的 bool 值进行区分。
 
-此 flag 继承自 ``nn.Layer`` ，因此可通过 ``model.train()`` 和 ``model.eval()`` 来全局切换所有 sublayers 的分支状态。
+此 flag 继承自 `nn.Layer` ，因此可通过 `model.train()` 和 `model.eval()` 来全局切换所有 sublayers 的分支状态。
 
 ## 七、非 forward 函数导出
```
Comment (DrRyanHuang) [inline docs/guides/jit/case_analysis_cn.md:277]: 下面的：
```
another_func.pdiparams        // 存放模型中所有的权重数据
another_func.pdimodel         // 存放模型的网络结构
another_func.pdiparams.info   // 存放额外的其他信息
```

我这里本地运行是：

`another_func.json` 和 `another_func.pdiparams`

切换成 PIR 之后，导出的格式变了吗?

File: docs/guides/jit/case_analysis_cn.md [hunk 9]
```diff
@@ -407,12 +407,12 @@ def forward(x):
         return outs
     ```
 
-> 因为框架底层的 ``DenseTensorArray = std::vector< DenseTensor >`` ，不支持两层以上 ``vector`` 嵌套
+> 因为框架底层的 `DenseTensorArray = std::vector<DenseTensor >` ，不支持两层以上 `vector` 嵌套
 
 ## 九、jit.save 与默认参数
 
 
-最后一步是预测模型的导出，Paddle 提供了 ``paddle.jit.save`` 接口，搭配 ``@to_static`` 可以导出预测模型。
```
Comment (DrRyanHuang) [inline docs/guides/jit/case_analysis_cn.md:415]: 看下面代码的注释，这里要不要说一下 to_static 的 Lazy 模式，只有在 save 导出或者运行的时候才会进行动转静

File: docs/guides/jit/case_analysis_cn.md (hunk 8 above)
Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:277]: `.pdimodel` 是什么鬼，我之前在动转静目录替换过 `.pdmodel`，没想到这里还藏了个 `.pdimodel`

File: docs/guides/jit/basic_usage_cn.md (hunk 1 above)
Comment (SigureMo) [reply docs/guides/jit/basic_usage_cn.md:226]: 这里是为啥？前向传播不包含 loss 计算吗？

File: docs/guides/jit/basic_usage_cn.md (hunk 4 above)
Comment (SigureMo) [reply docs/guides/jit/basic_usage_cn.md]: 按我理解是传递给 `paddle.jit.save` 的 `input_spec` 必须是 `None`，因为此时已经不确定具体传递给哪个函数了

`paddle.jit.to_static` 装饰时的必须详细为每一个函数指定 `input_spec`

File: docs/guides/jit/case_analysis_cn.md (hunk 6 above)
Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:49]: 其实我觉得倒也没必要，因为 `Placeholder` 确实在静态图里是更常见的术语，特别是在 TensorFlow 1.x 时期，改为中文反而看不懂

File: docs/guides/jit/case_analysis_cn.md (hunk 7 above)
Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:196]: 原理那边的最后，这篇在原理之后，跨页面引用目前没有太好的方式

File: docs/guides/jit/case_analysis_cn.md (hunk 8 above)
Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:277]:
```python
        with dygraph.guard():
            if use_pir_api():
                for tensor, value in zip(*concrete_program.parameters):
                    if not value.persistable:
                        continue
                    param_or_buffer_tensor = scope.var(value.name).get_tensor()
                    src_tensor = (
                        state_var_dict[tensor.name].value().get_tensor()
                    )
                    param_or_buffer_tensor._share_data_with(src_tensor)

            else:
                for param_or_buffer in concrete_program.parameters:
                    # share to scope
                    if param_or_buffer.type == core.VarDesc.VarType.VOCAB:
                        scr_tensor = param_or_buffer.value().get_map_tensor()
                        tgt_var = scope.var(param_or_buffer.name)
                        tgt_var.set_vocab(scr_tensor)
                    else:
                        param_or_buffer_tensor = scope.var(
                            param_or_buffer.name
                        ).get_tensor()
                        # src_tensor = param_or_buffer.value().get_tensor()
                        src_tensor = (
                            state_var_dict[param_or_buffer.name]
                            .value()
                            .get_tensor()
                        )
                        param_or_buffer_tensor._share_data_with(src_tensor)
                    # record var info
                    if param_or_buffer.name not in extra_var_info:
                        extra_info_dict = {}
                        if param_or_buffer.name in state_names_dict:
                            extra_info_dict['structured_name'] = (
                                state_names_dict[param_or_buffer.name]
                            )
                        extra_info_dict['stop_gradient'] = (
                            param_or_buffer.stop_gradient
                        )
                        if isinstance(param_or_buffer, EagerParamBase):
                            extra_info_dict['trainable'] = (
                                param_or_buffer.trainable
                            )
                        extra_var_info[param_or_buffer.name] = extra_info_dict
```

根据这段逻辑，PIR 已经不会再往 `extra_var_info` 塞东西了，所以没有 `.pdiparams.info` 文件，这我还是第一次知道，我一直以为 PIR 也会有这个文件

不过 `.pdmodel` 被 `.json` 取代是确定的

File: docs/guides/jit/case_analysis_cn.md (hunk 9 above)
Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:415]: 这个会不会没什么必要？

File: docs/guides/jit/basic_usage_cn.md (hunk 5 above)
Comment (SigureMo) [reply docs/guides/jit/basic_usage_cn.md:890]: > 这部分是不是应该修改一下，把参数传给 `save` 或者，给 `to_static` 加参数

？参数是用 `input_spec` 描述的啊，怎么能传给 `save`/`to_static`？

File: docs/guides/jit/basic_usage_cn.md (hunk 1 above)
Comment (DrRyanHuang) [reply docs/guides/jit/basic_usage_cn.md:226]: 主要是 `前向预测` 可能被误解为 `self.is_training=False` 的情况

File: docs/guides/jit/case_analysis_cn.md (hunk 8 above)
Comment (DrRyanHuang) [reply docs/guides/jit/case_analysis_cn.md:277]: 之前套件里面用自带的 `export.py` 导出静态图模型，静态图模型 save 后就会有这三个文件：
`another_func.pdiparams`
`another_func.pdimodel`
`another_func.pdiparams.info`

File: docs/guides/jit/basic_usage_cn.md (hunk 5 above)
Comment (DrRyanHuang) [reply docs/guides/jit/basic_usage_cn.md:890]: 不是，你康康这里：
```python
net = to_static(input_spec=[InputSpec(shape=[None, 10], name='x')])
```
`to_static` 里面没传模型

File: docs/guides/jit/case_analysis_cn.md (hunk 8 above)
Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:277]: `pdmodel`，哪来的 `pdimodel`

File: docs/guides/jit/basic_usage_cn.md (hunk 5 above)
Comment (SigureMo) [reply docs/guides/jit/basic_usage_cn.md:890]: 喔 这还真没看出来

File: docs/guides/jit/basic_usage_cn.md (hunk 1 above)
Comment (SigureMo) [reply docs/guides/jit/basic_usage_cn.md:226]: 那改成 `仅实现预测所需功能`？因为这里语义是不要包含仅训练所需的 loss

File: docs/guides/jit/case_analysis_cn.md (hunk 8 above)
Comment (DrRyanHuang) [reply docs/guides/jit/case_analysis_cn.md:277]: 康康套件这里的介绍
https://github.com/PaddlePaddle/PaddleDetection/tree/release/2.8.1/deploy#11-%E5%AF%BC%E5%87%BA%E6%A8%A1%E5%9E%8B

Comment (SigureMo) [reply docs/guides/jit/case_analysis_cn.md:277]: 这不都是 pdmodel 吗

**Review by DrRyanHuang — APPROVED** [review]
